src/single-spa-react.js

```diff
@@ -217,8 +217,15 @@ function defaultDomElementGetter(props) {
 }
 
 function reactDomRender({ opts, elementToRender, domElement, whenFinished }) {
-  if (opts.renderType === "createRoot") {
-    return opts.ReactDOM.createRoot(domElement).render(
+  if (
+    [
+      "createRoot",
+      "unstable_createRoot",
+      "createBlockingRoot",
+      "unstable_createBlockingRoot",
+    ].indexOf(opts.renderType) >= 0
+  ) {
+    return opts.ReactDOM[opts.renderType](domElement).render(
```

Comment:
honestly, I think just putting all the different options in here and removing the code below is the better way forward.

(it's how I originally wanted this code to be anyway, but there was some pushback on it so I did it this way)